### PR TITLE
Moveit bringup files

### PR DESCRIPTION
**Pull request by wuphilipp**
As per https://github.com/berkeleyopenarms/blue_core/pull/62

### CODE REVIEW

**Comment by methylDragon** [comment]
Should we put a run.launch.xml specific to this package that starts the controllers relevant to moveit? Otherwise the bringup here will have to depend on blue_bringup.

**Comment by wuphilipp** [comment]
@methylDragon 
Does this change do what you want?

**Comment by methylDragon** [comment]
Think this works fine (: :+1:

**Comment by methylDragon** [comment]
This PR requires https://github.com/berkeleyopenarms/blue_moveit/pull/5 to work for the right arm. And will need a similar PR to fix it for blue_left

(This is because the blue_side arg called in move_group.launch looks for blue_right or blue_left SRDFs and other associated yaml files.)

**Comment by wuphilipp** [comment]
@methylDragon I moved your changes from #5 over here. One difference I made was to always load the blue_full_moveit_controllers.yaml, which should simpler. I haven't tested this (it's late for us in the US), so if you have a change let me know if this works as intended! Otherwise, I will test and merge in the morning!

**Comment by methylDragon** [comment]
Don't see any issues with this. Only thing is that the bringup launchfile will throw warnings that no left_arm controller is found since it isn't loaded. That, and the param list will get filled with left_arm namespaced parameters that aren't in use. (That's expected behaviour if we run the right bringup launchfile.)

My arm works fine with this branch though, so that's good!

**Comment by wuphilipp** [comment]
I see, I went ahead and separated them. Hopefully, this gets rid of the warnings.

**Comment by wuphilipp** [comment]
@brentyi can we merge this? I think after merging we can do a separate PR which cleans the package up a bit.

**Comment by brentyi** [comment]
^Sorry for the delay on reviewing all your PRs, still waiting to get approved at work :(

**Review by brentyi — COMMENTED** [review]
Would it make sense to remove the "core robot" bringup stuff from these launch files? (eg the inclusion of `run.launch.xml` from `blue_bringup`)

It'd be a little more effort, but I might prefer it if running MoveIt on the robot came in two steps:
1. Run the core launch file from `blue_bringup`
2. Run the MoveIt stack from `blue_moveit_bringup`

Similarly, running MoveIt w/ Gazebo would be:
1. Run the core launch file from `blue_simulator`
2. Run the MoveIt stack from `blue_moveit_bringup`

This would be more effort to type, but I think this would be both clearer and more flexible. Features like the `on_stand` and `rosbridge_port` parameters implemented in `blue_core` wouldn't need to be reimplemented to this repo, etc. Users who want a single launch file to run everything could just make one that has `<include />` tags for both the core launch file and the corresponding MoveIt one, possibly with arguments like `param_file`, `version`, and `on_stand` pre-filled in.

**Comment by wuphilipp** [comment]
Ok, I agree. That will make things more clear and easier to maintain.

**Comment by wuphilipp** [comment]
@brentyi What do you think now?

File: blue_moveit_config/launch/move_group.launch [hunk 1]
```diff
@@ -32,7 +32,7 @@
   <!-- Trajectory Execution Functionality -->
   <include ns="move_group" file="$(find blue_moveit_config)/launch/trajectory_execution.launch.xml" if="$(arg allow_trajectory_execution)">
     <arg name="moveit_manage_controllers" value="true" />
-    <arg name="moveit_controller_manager" value="$(arg blue_side)" unless="$(arg fake_execution)"/>
+    <arg name="moveit_controller_manager" value="blue_full" unless="$(arg fake_execution)"/>
```
Comment (brentyi) [inline blue_moveit_config/launch/move_group.launch:35]: Why is this hardcoded to `blue_full`? We pass in a `blue_side` argument, but it seems like it's basically ignored?

File: blue_moveit_bringup/launch/left_moveit.launch [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    <arg name="allow_trajectory_execution" value="true"/>
+    <arg name="fake_execution" value="false"/>
+    <arg name="info" value="true"/>
+    <arg name="debug" value="$(arg debug)"/>
+    <arg name="blue_side" value="blue_left" />
+    <arg name="version" value="$(arg version)" />
+  </include>
+
+  <!-- Run Rviz and load the default config to see the state of the move_group node -->
+  <include file="$(find blue_moveit_config)/launch/moveit_rviz.launch">
```
Comment (brentyi) [inline blue_moveit_bringup/launch/left_moveit.launch]: Perhaps the rviz include can be separated/removed as well? It doesn't really seem core to running MoveIt

File: blue_moveit_bringup/launch/left_moveit.launch [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<launch>
+  <arg name="version" default="2" doc="1 or 2"/>
+  <!-- By default, we are not in debug mode -->
+  <arg name="debug" default="false" />
```
Comment (brentyi) [inline blue_moveit_bringup/launch/left_moveit.launch]: Maybe good to add a comment on what "debug mode" actually is

**Review by brentyi — COMMENTED** [review]
seems pretty much ready to merge!

File: blue_moveit_config/launch/move_group.launch (hunk 1 above)
Comment (wuphilipp) [reply blue_moveit_config/launch/move_group.launch:35]: Currently, the ```blue_side``` argument will only load in the specified moveit_sensor_manager and srdf. A move group for the full arm will be attempted. This was left over from when we had additional files which specified for each side. We can

1. Remove the reference to ```blue_side``` here
or 
2.  Add back in the files that use the ```blue_side``` argument.

**Comment by wuphilipp** [comment]
@brentyi  Any other changes we should make?

**Comment by brentyi** [comment]
What's the purpose of `gazebo_demo.launch`? Is a difference between that and just running `full.launch` from blue_gazebo and then `full_moveit.launch` from blue_moveit_bringup?

**Comment by wuphilipp** [comment]
Humm seems like it. I've gone ahead and remove the launch file.

**Comment by brentyi** [comment]
Okay, seems like we're almost there...

Is this a familiar error message?
image

I get it from `blue_moveit_bringup/launch/full_moveit.launch`, which I run after I start up Gazebo with `blue_gazebo/launch/full.launch`.

Ps -- also extremely minor but should we establish a standard for making these launch file names consistent? We currently have `full.launch` in blue_bringup and blue_gazebo, `full_moveit.launch` in blue_moveit_bringup, and `teleop_full.launch` in blue_teleop (granted, this one is internal).

**Comment by wuphilipp** [comment]
I can't seem to reproduce the error on my end. I can try testing it on a clean 18.04 installation tomorrow. I could be that you need install ```moveit_simple_controller_manager```. I noticed it was missing from the package.xml and just added it. 

As for the moveit launch file, I think maybe we should prepend with full/right/left just because it would be eaiser to tab complete. No strong opinions though.

**Comment by brentyi** [comment]
That fixed it!

- Naming: if these launch files are called `right_moveit`, `full_moveit`, etc, should the Gazebo ones be called `right_gazebo`, `full_gazebo`, etc? Same question for core.
- What's the preferred way to run rviz for messing with MoveIt? (the only way I figured out how to launch a pre-configured rviz instance was `roslaunch blue_moveit_config moveit_rviz.launch config:=true`, but that doesn't seem documented anywhere -- should `rviz.launch` in `blue_bringup` work?)

**Comment by wuphilipp** [comment]
What do you think about just naming the files ```full.launch``` etc? This might make things more confusing but the package the file is in could be enough to define its purpose.

For the preferred rviz visualization, I'm not to sure. Would you want to remove the other rviz launchs and add a parameter to the ```rviz.launch``` for Moveit?

**Pull request merged**